### PR TITLE
Update linter settings and update imports

### PR DESCRIPTION
**Pull request by Taztingo**
## Description

A lot of changes were made to support SDK v0.50 and we have not linted anything yet. This updates our linter settings and fixes the imports in our source. A separate PR will be made that lints the actual code.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

## Summary by CodeRabbit

- **Chores**
  - Optimized import ordering for maintainability and consistency.
  - Updated import paths and removed redundant imports for smoother module integration.

- **Documentation**
  - Updated CHANGELOG.md to reflect linter settings and import adjustments.

These changes aim to enhance code quality and maintain consistency in the project.

### CODE REVIEW

**Review by nullpointer0x00 — DISMISSED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -85,11 +85,29 @@ linters-settings:
 
           - cosmossdk.io/core
           - cosmossdk.io/errors
+          - cosmossdk.io/log
           - cosmossdk.io/math
           - cosmossdk.io/store
-          - github.com/cosmos/go-bip39
+          - cosmossdk.io/x/evidence
+          - cosmossdk.io/x/evidence/keeper
+          - cosmossdk.io/x/evidence/types
```
Comment (SpicyLemon) [inline .golangci.yml:93]: You should be okay to just have the root package here for these, e.g. `cosmossdk.io/x/evidence`. Basically, whatever's in `go.mod`.

I'm not 100% sure on this though, but it seems to work just fine to have `github.com/cosmos/cosmos-sdk` to cover all the packages contained in that library.

Comment (SpicyLemon) [reply .golangci.yml:93]: Actually. It's probably okay to leave it like this for now. Later, once the linter's fully passing, if we care, we can clean up some entries.

File: .golangci.yml [hunk 2]
```diff
@@ -138,14 +170,14 @@ linters-settings:
     sections:
       - standard
       - default
-      - prefix(github.com/tendermint)
+      - prefix(github.com/cometbft)
       - prefix(cosmossdk.io)
       - prefix(github.com/cosmos)
       - prefix(github.com/provenance-io)
       - blank
       - dot
```
Comment (SpicyLemon) [inline .golangci.yml:178]: I've been rethinking these settings. I wonder if it'd be easier on us if we just did

```
      - standard
      - default
      - prefix(github.com/provenance-io)
      - blank
      - dot
```

I don't know about your IDEs, but in IntelliJ, you can't define the auto-import settings to exactly match our current lint settings, so I'm constantly having to fix the imports in files after I make changes. Are our current settings a pain for you too?

**Review by SpicyLemon — APPROVED** [review]

File: .golangci.yml (hunk 2 above)
Comment (Taztingo) [reply .golangci.yml:178]: This hasn't caused too much of an issue for me with VSCode. It could be related to how your IDE is configured.

**Review by nullpointer0x00 — APPROVED** [review]

**Review by SpicyLemon — APPROVED** [review]

**Pull request merged**